src/Settings/DefaultPathManager.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (C) 2019-2020 Ashar Khan <ashar786khan@gmail.com>
+ *
+ * This file is part of CP Editor.
+ *
+ * CP Editor is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * I will not be responsible if CP Editor behaves in unexpected way and
+ * causes your ratings to go down and or lose any important contest.
+ *
+ * Believe Software is "Software" and it isn't immune to bugs.
+ *
+ */
+
+#include "Settings/DefaultPathManager.hpp"
+#include "Core/EventLogger.hpp"
+#include "Settings/SettingsManager.hpp"
+#include "generated/SettingsHelper.hpp"
+#include <QDebug>
+#include <QFileInfo>
+#include <QRegularExpression>
+
+QMap<QString, QString> DefaultPathManager::defaultPath;
+
+QString DefaultPathManager::defaultPathForAction(const QString &action)
+{
+    const auto result =
+        convertPath(SettingsManager::get(QString("Default Path/Action/%1/Uses").arg(action)).toString());
+    LOG_INFO(INFO_OF(action) << INFO_OF(result));
+    return result;
+}
+
+void DefaultPathManager::changeDefaultPathForAction(const QString &action, const QString &path)
+{
+    LOG_INFO(INFO_OF(action) << INFO_OF(path));
+
+    const auto settingsKey = QString("Default Path/Action/%1/Changes").arg(action);
+
+    if (!SettingsManager::contains(settingsKey, true))
+    {
+        qDebug() << "Unkown Action:" << action;
```

Comment:
Maybe remove this?

Reply:
Why? It's just like unknown settings. All actions are strings, it's very possible to have a typo. It's more obvious to show this in stderr.

Reply:
Ok